docs/excel/custom-functions-batching.md

```diff
@@ -0,0 +1,144 @@
+
+    Replace `TODO#2` with the following code:
+
+    ```js
+    if (!_isBatchedRequestScheduled) {
+        setTimeout(_makeRemoteRequest, 2000);
+        _isBatchedRequestScheduled = true;
+    }
+    ```
+
+    Finally, the code sample returns the Promise for that invocationEntry.
+
+* **TODO #3**
+
+    Before the third `TODO` item, you'll notice that the `_isBatchedRequestScheduled` flag is set to false, because there is nothing to batch until an invocation of a function takes place.
+
+    When an invocation happens, it will be passed to the `batch` array to await processing, as shown in the following diagram. In the following diagram, three invocations of the `=STOCK` function are issued in Excel, giving a stock ticker and expecting back a stock price. You'll note that the invocationEntries that are stored in the `batch` array contain both the raw information of the stock ticker string as well as the unresolved Promise.
+
+    ![Three calls of =STOCK function are issued and stored in the batch array.](../images/Batch1.svg "Batching diagram")
+
+    The third `TODO` item follows, with a new function `_makeRemoteRequest`.
+
+    First, the `_makeRemoteRequest` function creates a copy of the original `batch` and clears out the `batch` array and flips the `_isBatchScheduled` flag back to false. This copy is called `batchCopy` and contains the unresolved Promises from your invocation entries. Creating the copy enables you to build up another batch while you are waiting for a response from your API.
+
+    Next, the `_makeRemoteRequest` function creates the `requestBatch` array, which contains just the raw information from the invocationEntries that needs to be processed by the API.
+
+    To give a visual example, the following diagram shows how the information from the `batch` array has been copied into the `batchCopy` array and the stock ticker string information separated out to the `requestBatch` array. The `batch` array has been emptied and could continue to fill with requests in the meantime while the `requestBatch` is processing.
+
+    ![The batch array is copied and emptied.](../images/Batch2.svg "Batching diagram")
+
+    To fill in the third `TODO` item, add the following code.
+
+    ```js
+    function _makeRemoteRequest() {
+        // Copy the shared batch and allow the building of a new batch while you are waiting for a response.
+
+        var batchCopy = _batch.slice();
+        _batch = [];
+        _isBatchScheduled = false;
+        // Build a simpler request batch that only contains the arguments for each invocation.
+
+        var requestBatch = [];
+        for (var i = 0; i < batchCopy.length; i++) {
+            requestBatch[i] = {
+                "operation": batchCopy[i].operation,
+                "arguments": batchCopy[i].arguments
+            };
+    }
+    ```
+
+* **TODO #4**
+
+    In the fourth `TODO`, you'll continue filling in the `_makeRemoteRequest` function.
```

Comment:
To complete the fourth 'TODO', continue to fill in the '_makeRemoteRequest' function.